<a href="https://colab.research.google.com/github/minuuva/ds1002/blob/main/Minu_Choi_Due_4_Feb_at_10_pm.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## DS 1002 homework due Sunday Feb 4 at 10 pm



The notebook from Wednesday's class is in my GitHib repository, including the little script we wrote at the end to find mangos in a list of fruit.  The file is called `MorePythonBasics.ipynb`.

(By the way I wondered if the correct spelling was "mangos" or "mangoes" and I found [this](https://www.mango.org/blog-mangos-or-mangoes/)!  Who knew?  There's a National Mango Board!  And they're prepared to adjudicate this and other difficult mango-related questions.)




# **I worked with Oliver Andres**

### Problem 1:  finding fruit

Adapt the function we wrote in class so that it takes two arguments,

`fruit` and `hungryfor` ,

and then searches through the list `fruit` for the item `hungryfor`.  The line that calls the function should look like

`find_fruit(food, hungryfor)`

Make it print out the results in the same way, so for example if `food` is what we entered in class and `hungryfor = "tomatoes"` then the output will be

```python
No tomatoes here.
No tomatoes here.
No tomatoes here.
No tomatoes here.
No tomatoes here.
```
Test your code by changing the values of `food` and `hungryfor` and make sure the output is what you expect.  Show a few outputs, each in its own code chunk.



In [1]:
# put code here
food = ["apples", "bananas", "oranges", "grapes", "pears"]
hungryfor = "tomatoes"

In [2]:
# put your first function call here, and make more code chunks for additional function calls
def find_fruit(fruit, hungryfor):
  for item in fruit:
    if item == hungryfor:
      print(f"I found {hungryfor}!")
    else:
      print(f"No {hungryfor} here")
find_fruit(food, hungryfor)

No tomatoes here
No tomatoes here
No tomatoes here
No tomatoes here
No tomatoes here


In [3]:
food = ["cherries", "mangoes", "blueberries", "coconuts", "peaches"]
hungryfor = "coconuts"
find_fruit(food, hungryfor)

No coconuts here
No coconuts here
No coconuts here
I found coconuts!
No coconuts here


In [4]:
food = ["kiwis", "lemons", "durians", "dates", "limes"]
hungryfor = "lemons"
find_fruit(food, hungryfor)

No lemons here
I found lemons!
No lemons here
No lemons here
No lemons here


#### Notes
1. Once you have run a code chunk in the notebook, it will remember that it has been run.  You don't have to call the function in the same code chunk where the function is defined.  In other words, to call the function with new inputs you can simply make a new code chunk, change the inputs, and call the function.  *You don't have to retype the definition of the function each time.*

2. Make sure you clean things up when you're done, so you don't include old code that doesn't work or old incorrect outputs.


### Problem 2:  the `input()` function

(a) Programming gets way more fun when you start to interact with your own programs!  Here is a function that lets you start doing that.  Run the following code and see what happens:


In [5]:
input()
print("Hello there!")

hi
Hello there!


(b) As you just saw, the code waits for you to enter something before it continues.  If you want to actually use the input in the code, you should assign its value to a variable:

In [6]:
message = input()
print(message)

hi
hi


(c) Now you can start interacting.  Write some code that asks the user what they are hungry for (using a print statement), and then takes the user input, and then runs the `find_fruit` function from the previous code block using the user input as the `hungryfor` variable.  Test your code, and then include a few different outputs that demonstrate that the code works.

In [7]:
# put code here
print("What are you hungry for?")
hungryfor = input()
find_fruit(food, hungryfor)

What are you hungry for?
lemons
No lemons here
I found lemons!
No lemons here
No lemons here
No lemons here


In [8]:
# run it here, and add more code chunks for more runs
print("What are you hungry for?")
hungryfor = input()
find_fruit(food, hungryfor)

What are you hungry for?
kiwis
I found kiwis!
No kiwis here
No kiwis here
No kiwis here
No kiwis here


### Problem 3:  ratings  
Write a piece of code which prompts a user with the question "What is your rating? (1-10)" and then asks for user input.  Let's say the input is 3.  Once the input is received, print out a message that says.

"Your rating is *** ."

Obviously, your program should print the number of stars corresponding to the input!  If the input is 5, it should print

"Your rating is ***** ."

Show at least two outputs of your code.

In [9]:
# put code here
print("What is your rating? (1-10)")
rating = int(float(input()))
if rating < 1:
  print("invalid input.")
else:
  stars = '*' * rating
  print(f"Your rating is {stars} .")

What is your rating? (1-10)
7
Your rating is ******* .


In [10]:
print("What is your rating? (1-10)")
rating = int(float(input()))
if rating < 1 or rating > 10:
  print("invalid input.")
else:
  stars = '*' * rating
  print(f"Your rating is {stars} .")

What is your rating? (1-10)
4.2
Your rating is **** .


In [11]:
# you might have to copy the code to show another output
# (unless you figure out another way to do it)
print("What is your rating? (1-10)")
rating = int(float(input()))
if rating < 1 or rating > 10:
  print("invalid input.")
else:
  stars = '*' * rating
  print(f"Your rating is {stars} .")

What is your rating? (1-10)
11
invalid input.


You do not have to do this as part of the assignment, but one thing to think about with any code that has **interactivity** is that you want to make it **robust**.  This means you should imagine someone is *trying* to break your code, and you should try to make it as indestructible as possible.

What will happen to your code if someone enters 3.5?  Or -12?  Or "nonsense"?  Test it and see.  Think about how you might guard against errors.


***In an attempt to make the code as indestructible as possible, I used an if/else statement. This accounts for inputs that are out of the range (1-10) which will return "invalid input." Otherwise, the code will return the correct number of stars as the rating. Considering the fact that the system recognizes rating as a string, I first changed the type to a float value then to an integer. Therefore, as shown in the second block of code, when 4.2 is the input, it will return 4 stars. ***